torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -254,6 +262,10 @@ TORCH_LIBRARY(c10d, m) {
   m.def(
       "allreduce_",
       dispatch(c10::DispatchKey::CompositeExplicitAutograd, allreduce_));
+  m.def(
+      "traceable_allreduce",
+      torch::dispatch(
+          c10::DispatchKey::CompositeExplicitAutograd, traceable_allreduce));
```

Comment:
Is this what's triggering the issue with the meta registration?

Reply:
Yes, but i haven't thought too hard about the fix.  maybe we just need to write out derivative formulas for these ops.  I think we can revisit this once we get a better idea what apis we want.